### PR TITLE
[DEVHUB-1414] Reinstate ISR with error handling and on-demand revalidation

### PR DESCRIPTION
**Pull request by ajhenri**
This will reinstate ISR rendering for dynamic pages. Article pages are fully re-rendered while topic and topic content pages are re-rendered partially; the reason for this being that additional backend services will need to be created in Strapi ([DEVHUB-1429](https://jira.mongodb.org/browse/DEVHUB-1429)) to optimize these pages so that they can be rendered faster if server side rendered.

Additionally, this PR does the following:

- Significantly decreases build time (1m-2m)
- Adds error handling for when ISR fails. For example, if an article fails to render, the server will try and find the article within the pre-evaluated data. In the unlikely scenario that it is not there, it will fail gracefully with a 404, which is no different than current behavior when a DA publishes new content.
- A new revalidation endpoint is created so that individual articles can be refreshed on demand or alternatively, the whole site (this trigger will be added later once cron is deprecated).
- The drone cron will continue to operate until DEVHUB-1429 is completed.

### CODE REVIEW

**Review by mcolella14 — COMMENTED** [review]
Just realized you haven't added me as a reviewer yet so I won't fully review, but just had some questions and didn't want to forget them.

File: src/pages/api/revalidate-webhook.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        return res.json({ revalidated: true });
+    } catch (err) {
+        Sentry.captureException(err);
+        // If there was an error, Next.js will continue
+        // to show the last successfully generated page
+        return res.status(500).send('Error revalidating');
+    }
+};
+
+export default revalidateHandler;
```
[the remark this answers is not in the document]
Comment (ajhenri) [reply src/pages/api/revalidate-webhook.ts]: This request will be coming from the CMS, being created during the lifecycle hooks. See https://github.com/10gen/devhub-cms/pull/145/files

File: src/pages/api/revalidate-webhook.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+const revalidateHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+    const { body } = req;
+    if (!req.headers.authorization) return res.status(401);
+    const token = req.headers.authorization.replace('Bearer ', '').trim();
+    if (token !== process.env.REVALIDATE_TOKEN) return res.status(401);
+
+    try {
+        if (body.slug) {
+            await res.unstable_revalidate(`/developer${body.slug}`);
```
[the remark this answers is not in the document]
Comment (ajhenri) [reply src/pages/api/revalidate-webhook.ts:18]: @mcolella14 True. While it is a minor version up, I did try this and it was not a clean upgrade. There will be some migration work for this, specifically with middleware and request handling. https://nextjs.org/docs/upgrading#upgrading-to-122)

[the remark this answers is not in the document]
Comment (ajhenri) [reply src/pages/api/revalidate-webhook.ts:18]: Let me try and do this upgrade. Will follow up with another commit.

**Review by gfunk1230 — APPROVED** [review]

**Pull request merged**